### PR TITLE
[HttpKernel] Follow semantic versioning naming

### PR DESCRIPTION
**Pull request by 94noni**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | .
| Tickets       | .
| License       | MIT
| Doc PR        | .

Related to:

 - https://symfony.com/doc/current/contributing/community/releases.html
 - https://semver.org/spec/v2.0.0.html#semantic-versioning-200

imho a `release` version is more the full version than just the patch number of the version
of course feel free to close :)

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -76,10 +76,11 @@ abstract class Kernel implements KernelInterface, RebootableInterface, Terminabl
     private static $freshCache = [];
 
     public const VERSION = '6.0.0-DEV';
+    public const RELEASE_VERSION = self::VERSION;
     public const VERSION_ID = 60000;
     public const MAJOR_VERSION = 6;
     public const MINOR_VERSION = 0;
-    public const RELEASE_VERSION = 0;
```
Comment (jderusse) [inline src/Symfony/Component/HttpKernel/Kernel.php:82]: Sounds like a BC Break

**Review by derrabus — CHANGES_REQUESTED** [review]
I agree with @jderusse: this is a breaking change.

And tbh, I don't see that this change solves an actual problem. We might pick different constant names if we started Symfony from scratch today. But changing the names now won't do us any good, imho.

**Comment by 94noni** [comment]
@derrabus @jderusse hello!

> Sounds like a BC Break

it is yes, not sounds like; that is why v6.0 is targeted were BC can occurs, am i wrong?

> won't do us any good

I've found no usage of this constant in the internal code base that is why I proposed to change it, maybe you are talking of internal core team code logic?
obviously keeping the previous naming in case external code use it as it is public 

> solves an actual problem

it aims not to "solve a problem", but rather be more precise that is all, as describe in the PR desc :)

thank you,

**Comment by wouterj** [comment]
> it is yes, not sounds like; that is why v6.0 is targeted were BC can occurs, am i wrong?

There must always be an upgrade path for a BC break, meaning you should be able to mark your code fully 6.0 compatible on 5.4. We do not allow hard breaks.

> it aims not to "solve a problem", but rather be more precise that is all, as describe in the PR desc :)

Imho, theory only is not a great reason to cause more upgrade work for applications using Symfony. Note that this is not used in any Symfony internal code, but nothing is forbidding using this in an application. So we must assume applications use this in their code.

**Comment by derrabus** [comment]
> @derrabus @jderusse hello!
> 
> > Sounds like a BC Break
> 
> it is yes, not sounds like; that is why v6.0 is targeted were BC can occurs, am i wrong?

Sure, but just because we can break BC here doesn't mean we should. Usually a BC break is prepared by introducing a compatibility layer that allows a smooth upgrade. It has to be possible to create an application that works with 5.4 and 6.0 without any code change. In your case, this means that the `PATCH_VERSION` constant needs to be introduced with 5.4 already.

My main concern is that you've changed the semantics of the `RELEASE_VERSION` constant. That is a very sneaky change that might break apps in subtle ways. *If* we go down that road (and I'm still not convinced we should), I'd rather remove that constant completely than change its semantics.

> > won't do us any good
> 
> I've found no usage of this constant in the internal code base that is why I proposed to change it, maybe you are talking of internal core team code logic?

I'm first and foremost talking about existing Symfony applications that might rely on the presence and current semantics of those constants. Apart from that, our internal release script that bumps those constants whenever we do a release would need to be adjusted as well.

**Comment by ro0NL** [comment]
just renaming RELEASE to PATCH sounds reasonable (with migration), given MAJOR.MINOR.PATCH is perhaps defacto terminology

**Comment by wouterj** [comment]
Actually, I'm :-1: for both ways (repurposing or renaming `RELEASE_VERSION`). Any deprecation/BC break adds more time to upgrade Symfony, so we must do a best effort to have the least number of them each release. As this doesn't solve a real problem, let's close.

If there is a real problem with this constant name, we can of course reconsider in the future.

**Comment by 94noni** [comment]
@wouterj I understood your maintainer POV here

but as I wrote, there is no `problem` to solve, just wording/naming improvement to map "reality"

just arriving inside this class from outside and reading constant `Kernel::RELEASE_VERSION`
One imagined more its value to `X.Y.Z` rather than `Z` I may think

---

anyhow, thanks for the explanations above and your time :)

**Pull request closed without merging**